runtime/recipe/type-checker.js

```diff
@@ -113,8 +113,8 @@ class TypeChecker {
     // guarantees that the view's type will be preserved, and that the fact
     // that the type comes from a view rather than a connection will also
     // be preserved.
-    let superDirection = superclass.connection ? superclass.connection.direction : 'inout';
-    let subDirection = subclass.connection ? subclass.connection.direction : 'inout';
+    let superDirection = superclass.direction || (superclass.connection ? superclass.connection.direction : 'inout');
```

Comment:
Needs @dstockwell review as I'm too new to this part of the code to know whether this perhaps weakens the conditions the comment above this describes. It did prompt me to spend some time reading code re: connections.

Reply:
absolutely, i'll wait for Doug's review.